sky/authentication.py

```diff
@@ -292,13 +237,15 @@ def setup_gcp_authentication(config):
 
     # OS Login is not enabled for the project. Add the ssh key directly to the
     # metadata.
+    # TODO(zhwu): Use cloud init to add ssh public key, to avoid the permission
+    # issue.
     project_keys = next(
         (item for item in project['commonInstanceMetadata'].get('items', [])
          if item['key'] == 'ssh-keys'), {}).get('value', '')
     ssh_keys = project_keys.split('\n') if project_keys else []
 
     # Generating ssh key if it does not exist
-    _, public_key = get_or_generate_keys(private_key_path, public_key_path)
+    _, public_key = get_or_generate_keys()
```

Comment:
Can we use the return values from the call to `get_or_generate_keys()` at L137 instead of invoking it again here?

Reply:
Done. Thanks!